website/sphinx/apidoc.sh

```diff
@@ -21,6 +21,9 @@ echo "Installing dependencies for generating Python API docs"
 $pythonCmd -m pip install -e $MLOS_ROOT/source/Mlos.Python/
 $pythonCmd -m pip install sphinx sphinx_rtd_theme numpydoc matplotlib
 
+# Work around some issues with python not picking up the system installed package sometimes.
+$pythonCmd -m pip install 'kiwisolver>=1.0.2'
```

Comment:
Addresses an issue I had while trying this in my WSL environment instead of docker where the `python-kiwisolver` package had been pulled in by `python-matplotlib` and the additional `python3.7` setup wasn't finding it.

Reply:
I'm not sure I understand that comment. It might be that you have an old/incomplete installation of matplotlib somewhere?

Reply:
Very possible.  Without that, I was getting this error:

```
Running Sphinx v3.2.1

Extension error:
Could not import extension matplotlib.sphinxext.plot_directive (exception: No module named 'kiwisolver')
```

Poking around I see this:
```
$ python3.7 -m pip show matplotlib kiwisolver
Name: matplotlib
Version: 3.3.2
Summary: Python plotting package
Home-page: https://matplotlib.org
Author: John D. Hunter, Michael Droettboom
Author-email: matplotlib-users@python.org
License: PSF
Location: /home/bpkroth/.local/lib/python3.7/site-packages
Requires: kiwisolver, python-dateutil, pyparsing, pillow, numpy, certifi, cycler
Required-by: seaborn
---
Name: kiwisolver
Version: 1.0.1
Summary: A fast implementation of the Cassowary constraint solver
Home-page: https://github.com/nucleic/kiwi
Author: The Nucleic Development Team
Author-email: sccolbert@gmail.com
License: UNKNOWN
Location: /usr/lib/python3/dist-packages
Requires:
Required-by: matplotlib

$ dpkg -l | grep kiwisolver
ii  python3-kiwisolver                   1.0.1-3build1                       amd64        fast implementation of the Cassowary constraint solver - Python 3.X
```

Possibly the better thing to do is add it to the previous line as

```
$pythonCmd -m pip install --upgrade sphinx sphinx_rtd_theme numpydoc matplotlib kiwisolver
```

Reply:
Of course, in the future we can switch to conda and consider this moot, but for now ...

Reply:
ah so it's system python polluting the environment. I'm curious why installing matplotlib doesn't fix that, but if you use system python, all bets are off. Sure, feel free to keep it for now and we'll move to conda later.

Reply:
Tried to consolidate to the latter form for now, and ended up with another dependency that it ran into issues with.
Thoughts?

Reply:
on the CI or locally? I think ti's fine as long as it works on CI, locally you seem to have some unrelated issues.

Reply:
I mean, I would prefer that the scripts work both locally and in CI, but agreed on there being something funny when the system python gets in the way.
As discussed, doing this all via docker/conda is probably the more supportable path forward.